src/ripple/shamap/impl/SHAMapTreeNode.cpp

```diff
@@ -79,8 +79,10 @@ SHAMapTreeNode::SHAMapTreeNode (Blob const& rawNode,
 {
     if (format == snfWIRE)
     {
-        Serializer s (rawNode);
-        int type = s.removeLastByte ();
+        assert (!rawNode.empty ());
+        Serializer s (rawNode.begin (),
+                      rawNode.empty () ? rawNode.end () : rawNode.end () - 1);
+        int type = rawNode.empty () ? -1 : rawNode.back ();
```

Comment:
Hmmm.  I like guarding against bad iterators.  That's a good thing.

What I'm noticing is that if `rawNode.empty()` is true we eventually get around to setting `type` to `-1` which causes a `throw std::runtime_error(...)`.  Since you're going to throw eventually anyway if `rawNode.empty()`, you could throw earlier in the code (replace the `assert` with a `throw`) and dispense with the ternary operators.  Seems like it might be a win.  Thoughts?